Lib/test/support/__init__.py

```diff
@@ -945,6 +945,27 @@ def check_sizeof(test, o, size):
             % (type(o), result, size)
     test.assertEqual(result, size, msg)
 
+def subTests(arg_names, arg_values, /, *, _do_cleanups=False):
+    """Run multiple subtests with different parameters.
+    """
+    if isinstance(arg_names, str):
+        arg_names = arg_names.split(',')
+    def decorator(func):
```

Comment:
With the current PR, if you use a generator for `arg_values`, a second run of the test will silently do nothing :skull_and_crossbones:
```suggestion
    arg_values = tuple(arg_values)
    def decorator(func):
```

Reply:
Done.